plutus-tx/src/PlutusTx/Blueprint/Schema.hs

```diff
@@ -0,0 +1,417 @@
+          ]
+    SchemaBuiltInData title description comment ->
+      Aeson.object
+        $ catMaybes
+          [ fmap ("title" .=) title
+          , fmap ("description" .=) description
+          , fmap ("$comment" .=) comment
+          ]
+    SchemaBuiltInUnit title description comment ->
+      dataType title description comment "#unit"
+    SchemaBuiltInBoolean title description comment ->
+      dataType title description comment "#boolean"
+    SchemaBuiltInInteger title description comment ->
+      dataType title description comment "#integer"
+    SchemaBuiltInBytes title description comment ->
+      dataType title description comment "#bytes"
+    SchemaBuiltInString title description comment ->
+      dataType title description comment "#string"
+    SchemaBuiltInPair title description comment schema1 schema2 ->
+      Aeson.object
+        $ catMaybes
+          [ Just $ "dataType" .= ("#pair" :: String)
+          , Just $ "fst" .= schema1
+          , Just $ "snd" .= schema2
+          , fmap ("title" .=) title
+          , fmap ("description" .=) description
+          , fmap ("$comment" .=) comment
+          ]
+    SchemaBuiltInList title description comment schema ->
+      Aeson.object
+        $ catMaybes
+          [ Just $ "dataType" .= ("#list" :: String)
+          , Just $ "items" .= schema
+          , fmap ("title" .=) title
+          , fmap ("description" .=) description
+          , fmap ("$comment" .=) comment
+          ]
+    SchemaOneOf schemas -> Aeson.object ["oneOf" .= schemas]
+    SchemaAnyOf schemas -> Aeson.object ["anyOf" .= schemas]
+    SchemaAllOf schemas -> Aeson.object ["allOf" .= schemas]
+    SchemaNot schema -> Aeson.object ["not" .= schema]
+    SchemaRef ref -> Aeson.object ["$ref" .= ("#/definitions/" ++ ref)]
+   where
+    dataType :: Maybe String -> Maybe String -> Maybe String -> String -> Aeson.Value
+    dataType title description comment ty =
+      Aeson.object
+        $ catMaybes
+          [ Just $ "dataType" .= ty
+          , fmap ("title" .=) title
+          , fmap ("description" .=) description
+          , fmap ("$comment" .=) comment
+          ]
+
+type HasSchemaDefinition :: forall a. a -> [a] -> Constraint
+type family HasSchemaDefinition n xs where
+  HasSchemaDefinition n '[] =
+    GHC.TypeError
+      ( GHC.ShowType n
+          GHC.:<>: GHC.Text " schema was not found in the list of schema definitions."
+      )
+  HasSchemaDefinition x (x ': xs) = ()
+  HasSchemaDefinition x (_ ': xs) = HasSchemaDefinition x xs
+
+schemaRef :: forall typ schemas. (Typeable typ, HasSchemaDefinition typ schemas) => Schema
+schemaRef = SchemaRef $ show $ typeRep $ Proxy @typ
+
+----------------------------------------------------------------------------------------------------
+-- Modifiers ---------------------------------------------------------------------------------------
+
+setTitle :: String -> Schema -> Schema
+setTitle title = \case
```

Comment:
Similar point - it would be better to include this for review when there exists code that uses it. The usefulness of this is questionable since one should set their blueprint title when creating the blueprint, and rarely need to reset it afterwards.

Reply:
I will remove these modifiers as they aren't used in the test and arguably are premature.